include/dlaf/util_macros.h

```diff
@@ -0,0 +1,23 @@
+//
+// DLAF
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+#include <type_traits>
+
+/// @file
+
+/// Allows to hide `std::enable_if` expressions from generated documentation.
+/// If `BooleanExpr` is true it results in `Type`, otherwise it is empty (useful for exploiting SFINAE).
+#ifdef DLAF_DOXYGEN
+#define RETURN_TYPE_IF(Type, BooleanExpr) Type
+#else
+#define RETURN_TYPE_IF(Type, BooleanExpr) std::enable_if_t<BooleanExpr, Type>
```

Comment:
Does not always work. E.g.
```
template <class T>
RETURN_TYPE_IF(std::array<T, 2>, std::is_integral<T>::value) foo(T a);
```
(Parenthesis are not an option here.)

Reply:
As discussed, we opt for a simple solution: keep manually separated the two signatures.

Just for reference, there is a solution for this problem with comma and macros when parenthesis cannot be used. (https://stackoverflow.com/a/13842784)